Extensions/Multiplayer/multiplayertools.ts

```diff
@@ -774,15 +677,27 @@ namespace gdjs {
           let heartbeatUrl = `${rootApi}/play/game/${gameId}/public-lobby/${_lobbyId}/action/heartbeat`;
           headers['Authorization'] = `player-game-token ${playerToken}`;
           heartbeatUrl += `?playerId=${playerId}`;
+          const players = gdjs.multiplayerMessageManager.getConnectedPlayers();
           await fetch(heartbeatUrl, {
             method: 'POST',
             headers,
+            body: JSON.stringify({
+              players,
+            }),
           });
         }, heartbeatInterval);
       }
 
       // If we are connected to players, then the game can start.
       logger.info('Lobby game has started.');
+      // In case we're joining an existing lobby, read the saved messages to catch-up with the game state.
+      gdjs.multiplayerMessageManager.handleUpdateGameMessagesSaved(
+        runtimeScene
+      );
+      gdjs.multiplayerMessageManager.handleUpdateSceneMessagesSaved(
+        runtimeScene
+      );
```

Comment:
I would hide `handleUpdateSceneMessagesSaved` and `handleUpdateGameMessagesSaved` from the interface of MultiplayerMessageManager, and have instead a single function `handleSavedUpdateMessages`.
No reason the caller is able to call this for the scene and not the game or vice versa.

Reply:
done